src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,50 +1,154 @@
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index >= 0 && index <= count) {
+            if (index == count) {
+                add(value);
+            } else {
+                Node<T> nextNode = getNodeByIndex(index);
+                Node<T> prewNode = nextNode.prev;
+                Node<T> newNode = new Node<>(value, nextNode, prewNode);
+                if (prewNode == null) {
+                    firstNode = newNode;
+                } else {
+                    prewNode.next = newNode;
+                }
+                nextNode.prev = newNode;
+                count++;
+            }
+        } else {
+            throw new IndexOutOfBoundsException();
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T item : list) {
+            add(item);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (indexIsExist(index)) {
+            Node<T> temp = getNodeByIndex(index);
+            return temp.item;
+        }
+        throw new IndexOutOfBoundsException();
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (index < count && index >= 0) {
+            Node<T> temp = getNodeByIndex(index);
+            T oldValue = temp.item;
+            temp.item = value;
+            return oldValue;
+        }
+        throw new IndexOutOfBoundsException();
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (indexIsExist(index)) {
+            Node<T> temp = getNodeByIndex(index);
+            removeThis(temp);
+            return temp.item;
+        }
+        throw new IndexOutOfBoundsException();
     }
 
     @Override
     public boolean remove(T t) {
+        Node<T> temp = firstNode;
+        int i = 0;
+        while (i != count) {
+            if (temp.item == t || temp.item.equals(t)) {
+                removeThis(temp);
+                return true;
+            }
+            temp = temp.next;
+            i++;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return count;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return count == 0;
+    }
+
+    public void removeThis(Node<T> tempNode) {
+        if (tempNode.prev == null) {
+            firstNode = tempNode.next;
+        } else {
+            tempNode.prev.next = tempNode.next;
+        }
+        if (tempNode.next == null) {
+            lastNode = tempNode.prev;
+        } else {
+            tempNode.next.prev = tempNode.prev;
+        }
+        count--;
+    }
+
+    private boolean indexIsExist(int index) {
+        return index >= 0 && index < count;
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        Node<T> temp = firstNode;
+        int i = 0;
+        while (i != index) {
```

Comment:
in this method let's start moving from the first Entry OR from the last one
depending on whether index is closer to the start or to the end’